### PR TITLE
Simplify the JSON response

### PR DESCRIPTION
**Pull request by chrismwendt**
This simplifies the JSON response so it's no longer tied to the structure that the HTML templates expect.

@asfaltboy @idank Once this is merged in (to `asfaltboy/explainshell`), https://github.com/idank/explainshell/pull/125 should automatically update and be ready to merge 👌

### CODE REVIEW

File: explainshell/web/views.py [hunk 1]
```diff
@@ -18,63 +18,82 @@ def index():
...
-            kwargs['e'] = repr(kwargs['e'])
-        return jsonify(status=status, **kwargs)
+def merge_lists(l1, l2, key):
+    merged = {}
+    for item in l1+l2:
+        if item[key] in merged:
+            merged[item[key]].update(item)
+        else:
+            merged[item[key]] = item
+    return [val for (_, val) in merged.items()]
```
Comment (asfaltboy) [inline explainshell/web/views.py]: can just `return merged.values()`

File: explainshell/web/views.py [hunk 2]
```diff
@@ -18,63 +18,82 @@ def index():
...
-def formatted_response(fmt, status, **kwargs):
-    if fmt == 'html':
-        return render_template(STATUS_TYPE_TEMPLATES[status], **kwargs)
-    else:
-        if 'e' in kwargs:
-            kwargs['e'] = repr(kwargs['e'])
-        return jsonify(status=status, **kwargs)
+def merge_lists(l1, l2, key):
+    merged = {}
+    for item in l1+l2:
```
Comment (asfaltboy) [inline explainshell/web/views.py]: can use `itertools.chain` to avoid copying the lists into a 3rd lists

File: explainshell/web/views.py (hunk 1 above)
Comment (asfaltboy) [inline explainshell/web/views.py]: can just `return merged.values()`

File: explainshell/web/views.py [hunk 3]
```diff
@@ -18,63 +18,82 @@ def index():
...
-def explain_api():
-    return generic_explain('json')
+def explain_json():
+    result = explain()
+    if result['status'] is not 'success':
+        if 'e' in result:
+            result['e'] = repr(result['e'])
+        return jsonify(result)
+    else:
+        helpItemsDicts = [{ 'helpHTML': helpItem[0], 'helpclass': helpItem[1] } for helpItem in result['helptext']]
```
Comment (asfaltboy) [inline explainshell/web/views.py]: can be written like so for clarity:
```
        helpItemsDicts = [{
            'helpHTML': help_content, 'helpclass': help_class
        } for help_content, help_class in result['helptext']]
```

File: explainshell/web/views.py [hunk 4]
```diff
@@ -18,63 +18,82 @@ def index():
...
+            del d['helpclass']
+        return jsonify({ 'matches': matches })
 
 @app.route('/explain')
-def explain():
-    return generic_explain('html')
+def explain_html():
+    result = explain()
+    if result['status'] is 'success':
+        return render_template('explain.html', **result)
```
Comment (asfaltboy) [inline explainshell/web/views.py]: I actually prefer the previously used mapping `STATUS_TYPE_TEMPLATES` to a long list of if clause. This function can then be written like so:
```
def explain_html():
    result = explain()
    if result['status'] not in STATUS_TYPE_TEMPLATES:
        # TODO: log a warning somewhere ...
        return redirect('/')
    return render_template(STATUS_TYPE_TEMPLATES[result['status']], **result)
```

**Review by asfaltboy — CHANGES_REQUESTED** [review]
Overall looks really nice to me! Added a bunch of small comments below, thanks for the PR !

**Comment by chrismwendt** [comment]
Thanks for the tips! I updated this PR.

**Review by asfaltboy — APPROVED** [review]

**Pull request merged**